tests/generic_config_updater/test_mmu_dynamic_threshold_config_update.py

```diff
@@ -0,0 +1,117 @@
+    finally:
+        delete_checkpoint(duthost)
+
+
+def ensure_application_of_updated_config(duthost, value, pg_lossless_profiles):
+    """
+    Ensures application of the JSON patch config update by verifying dynamic threshold value presence in DB
+
+    Args:
+        duthost: DUT host object
+        value: expected value of dynamic threshold
+        pg_lossless_profiles: all pg_lossless buffer profiles stored on the device
+    """
+    def _confirm_value_in_appl_db_and_asic_db():
+
+        for pg_lossless_profile in pg_lossless_profiles:
+            # Retrieve dynamic_th from APPL_DB
+            dynamic_th_in_appl_db = duthost.shell("sonic-db-cli APPL_DB hget BUFFER_PROFILE_TABLE:{} dynamic_th".format(pg_lossless_profile))["stdout"]
+            if dynamic_th_in_appl_db != value:
+                return False
+
+        # Retrieve dynamic_th from ASIC_DB
+        ingress_lossless_pool_oid = duthost.shell("sonic-db-cli COUNTERS_DB hget COUNTERS_BUFFER_POOL_NAME_MAP ingress_lossless_pool")["stdout"]
+        buffer_pool_keys = duthost.shell("redis-cli -n 1 KEYS ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_PROFILE:oid*")["stdout_lines"]
+
+        for buffer_pool in buffer_pool_keys:
+            pool_oid = duthost.shell("sonic-db-cli ASIC_DB hget {} SAI_BUFFER_PROFILE_ATTR_POOL_ID".format(buffer_pool))["stdout"]
+
+            if pool_oid == ingress_lossless_pool_oid:
+                xoff_val = duthost.shell("sonic-db-cli ASIC_DB hget {} SAI_BUFFER_PROFILE_ATTR_XOFF_TH".format(buffer_pool))["stdout"]
+                dynamic_th_in_asic_db = duthost.shell("sonic-db-cli ASIC_DB hget {} SAI_BUFFER_PROFILE_ATTR_SHARED_DYNAMIC_TH".format(buffer_pool))["stdout"]
+                # Dynamic threshold values are a mismatch for pg_lossless profiles
+                if dynamic_th_in_asic_db != value and len(xoff_val) > 0:
+                    return False
+
+        return True
+
+    pytest_assert(
+        wait_until(READ_APPL_DB_TIMEOUT, READ_APPL_DB_INTERVAL, 0, _confirm_value_in_appl_db_and_asic_db),
+        "ASIC_DB or APPL_DB does not properly reflect new dynamic threshold expected value: {}".format(value)
+    )
+
+def get_pg_lossless_profiles(duthost):
+    """
+    Retrieves all pg_lossless buffer profiles that are present on the device. Ex. pg_lossless_100000_40m_profile
+
+    Args:
+    duthost: DUT host object
+    """
+    pg_lossless_profiles_str = duthost.shell("redis-cli -n 0 KEYS *BUFFER_PROFILE_TABLE:pg_lossless*")["stdout_lines"]
+    pg_lossless_profiles_lst = []
+
+    for pg_lossless_profile_str in pg_lossless_profiles_str:
+        pg_lossless_profile = pg_lossless_profile_str[21:]
+        pg_lossless_profiles_lst.append(pg_lossless_profile)
+
+    return pg_lossless_profiles_lst if len(pg_lossless_profiles_lst) > 0 else None
+
+@pytest.mark.parametrize("operation", ["replace"])
+def test_dynamic_th_config_updates(duthost, ensure_dut_readiness, operation):
+    pg_lossless_profiles = get_pg_lossless_profiles(duthost)
```

Comment:
Can we configure pg lossless profile if it's not already present on device so we don't rely on a certain configuration to begin with?

Reply:
The whole point of this test is to alter dynamic threshold values for pg_lossless profiles. If the switch does not contain pg_lossless profiles, we won't be modifying any dynamic thresholds

Reply:
Oh I see what you mean